Sources/MapboxMaps/Foundation/MapView.swift

```diff
@@ -618,6 +620,16 @@ open class MapView: UIView {
         cameraAnimatorsRunnerEnablable.isEnabled = true
 
         updateDisplayLinkPreferredFramesPerSecond()
+
+        // this will make sure that display link is only running on an active scene in foreground,
+        // preventing metal view drawing on background if the view is added to window not on foreground
+        if #available(iOS 13, *), let scene = window.windowScene, scene.activationState != .foregroundActive {
+            displayLink.isPaused = true
+        }
+        if UIApplication.shared.applicationState != .active {
+            displayLink.isPaused = true
+        }
```

Comment:
This as a method didn't look that good, made it work with some extensions

Reply:
Actually, not the implementation itself that motivated me to extract the method but the need to have a separate function. It seems like a complete logic unit behind this and it would look a bit better then, as for me.

Reply:
Good point about it being a complete logic unit! Changed it to be a method.